src/tracks/playabletrack/wavetrack/ui/WaveClipUtilities.cpp

```diff
@@ -29,7 +30,7 @@ void findCorrection(const std::vector<sampleCount> &oldWhere, size_t oldLen,
    const double denom = oldWhereLast - oldWhere0;
 
    // What sample would go in where[0] with no correction?
-   const double guessWhere0 = t0 * rate;
+   const double guessWhere0 = t0 * sampleRate / stretchRatio;
```

Comment:
near the end of this function `findCorrection`, clean up two things please:  use `std::clamp` and `assert`